---
title: "Stromal Analysis"
output: html_notebook
---

```{r message=FALSE, warning=FALSE}
suppressWarnings(library(Seurat))
suppressWarnings(library(scran))
suppressWarnings(library(scDblFinder))
suppressWarnings(library(tidyverse))
suppressWarnings(library(RColorBrewer))
suppressWarnings(library(openxlsx))
suppressWarnings(library(circlize))
suppressWarnings(library(gtools))
suppressWarnings(library(ggExtra))
suppressWarnings(library(ggridges))
suppressWarnings(library(patchwork))

projectDir <- "."
projectPath <- file.path(projectDir)
outputPath <- file.path(projectPath)
dataDir <- "data"
dataPath <- file.path(projectPath,dataDir)
prefix <- "Stromal"

dir.create(file.path(outputPath),recursive = T)

setwd(outputPath)

color.list <- c("#ebac23", "#b80058", "#008cf9",
                "#006e00", "#00bbad", "#d163e6",
                "#b24502", "#ff9287", "#5954d6",
                "#00c6f8", "#878500", "#00a76c",
                "#bdbdbd", "#846b54",
                brewer.pal(12, "Paired"),
                brewer.pal(12, "Set3"),
                brewer.pal(8, "Pastel2"),
                colorRampPalette(c("grey20","grey70"))(4))
```

# Load Counts

```{r}
sc <- Read10X(file.path(dataPath,"stromal.counts"))
sc <- CreateSeuratObject(
  counts = sc,
  assay = "RNA",
  project = "Stromal",
  names.field = c(1,2),
  names.delim = "_",
  min.cells = 0,
  min.features = 0
)
sc@meta.data$SampleName <- sc@meta.data$orig.ident

table(sc$SampleName)
```

# Filtering

## Filtering by sequencing depth

```{r}
minDepth <- 2000
maxDepth <- 40000
```

```{r }
sc@meta.data %>%
    ggplot( aes(x=nCount_RNA,y=SampleName)) +
    geom_density_ridges(aes(fill = SampleName),alpha=.5) +
    scale_fill_manual(values = color.list) +
    geom_jitter(height = 0.2, alpha = 0.01) +
    geom_vline(xintercept = minDepth, color="red") +
    geom_vline(xintercept = maxDepth, color="red") +
    ggtitle("Sequencing Depth Density Plot") +
    coord_cartesian(xlim=c(0,maxDepth*1.1)) +
    theme(legend.position = "none")

########### Filter By Total Counts ############
sc@meta.data$filter_by_total_counts <- sc$nCount_RNA > minDepth & sc$nCount_RNA < maxDepth  

as.data.frame(table(sc@meta.data[,c("filter_by_total_counts","SampleName")])) %>% pivot_wider(names_from = filter_by_total_counts, values_from = Freq)
```

### Filtering by genes detected

```{r}
minGenesDetected <- list("CD63_WT" = 800,
                         "CD63_KO" = 1000)
```

```{r }
sc@meta.data %>%
  ggplot(aes(x=nFeature_RNA,y=SampleName)) +
  geom_density_ridges(aes(fill = SampleName),alpha=.5) +
  scale_fill_manual(values = color.list) +
  geom_jitter(height = 0.2, alpha = 0.01) +
  geom_vline(xintercept = minGenesDetected$CD63_WT, color="red") +
  ggtitle("Genes Detected Density Plot") +
  theme(legend.position = "none")

sc@meta.data$filter_by_expr_features <- ifelse(
  sc$SampleName == "CD63_KO",
  sc$nFeature_RNA > minGenesDetected$CD63_KO,
  sc$nFeature_RNA > minGenesDetected$CD63_WT
)

as.data.frame(table(sc@meta.data[,c("filter_by_expr_features","SampleName")])) %>% pivot_wider(names_from = filter_by_expr_features, values_from = Freq)
```


### Filter by MT content

```{r}
mtGenes <- grep("mt-",rownames(sc),value = T)

mtCounts <- Matrix::colSums(sc@assays$RNA@counts[mtGenes,])

sc@meta.data$pct_counts_MT <- mtCounts * 100 / sc$nCount_RNA
```

```{r}
maxMT <- 10
```

```{r }
sc@meta.data %>%
ggplot(aes(x=pct_counts_MT,y=SampleName)) +
  geom_density_ridges(aes(fill=SampleName),alpha=0.5) +
  scale_fill_manual(values = color.list) +
  geom_jitter(height = 0.2, alpha = 0.01) +
  geom_vline(xintercept = maxMT, color="red") +
  ggtitle("MT Content Density Plot") +
  theme(legend.position = "none")

sc@meta.data$filter_by_MT_features <- sc$pct_counts_MT < maxMT

as.data.frame(table(sc@meta.data[,c("filter_by_MT_features","SampleName")])) %>% pivot_wider(names_from = filter_by_MT_features, values_from = Freq)
```

### Filter by Gene Expression Complexity

```{r}
topComplex <- function (x,t) {
  x <- sort(x,decreasing = T)
  return(sum(x[seq(t)],na.rm = T)*100/sum(x,na.rm = T))
}
sc@meta.data$percent.top_50 <- apply(sc@assays$RNA@counts,2,topComplex,t=50)
```

```{r}
maxPct <- 65
```

```{r }
sc@meta.data %>%
ggplot(aes(x=percent.top_50,y=SampleName)) +
  geom_density_ridges(aes(fill=SampleName)) +
  geom_jitter(height = 0.2, alpha = 0.01) +
  geom_vline(xintercept = maxPct, color="red") +
  scale_fill_manual(values = c(color.list,color.list)) +
  ggtitle("UMIs in Top 50 Genes Density Plot") +
  theme(legend.position = "none")

sc@meta.data$filter_by_Top50 <- sc$percent.top_50 < maxPct

as.data.frame(table(sc@meta.data[,c("filter_by_Top50","SampleName")])) %>% pivot_wider(names_from = filter_by_Top50, values_from = Freq)
```

### Filtering Summary

```{r }
######### Set Manual Filtering ##########
sc@meta.data <- sc@meta.data %>%
  mutate(manualFilter = 
  # sufficient features (genes)
  filter_by_expr_features &
  # sufficient molecules counted
  filter_by_total_counts &
  # remove cells with unusual number of reads in MT genes
  filter_by_MT_features &
  # remove cells with low gene complexity
  filter_by_Top50
)

df <- sc@meta.data %>%
  dplyr::select(SampleName,
                nCount_RNA,
                nFeature_RNA,
                contains("filter_by"),
                manualFilter)

df <- df %>% pivot_longer(
  cols = contains("filter",ignore.case = T),
  names_to = "Filter", values_to = "PASS")

ggplot(df,aes(x=log10(nCount_RNA),y=nFeature_RNA)) +
  geom_point(aes(color=PASS)) +
  facet_wrap("Filter")
```

```{r}
sc <- sc[,sc$manualFilter]
sc
```

# Integration by Sample using CCA

```{r}
cellSet <- "Stromal"
subsetName <- "Integrated"
```

```{r}
nfeatures <- 1000
ress <- c(0.1)
npcs <- 20

bySample <- SplitObject(sc, split.by = "SampleName")

# Integration via CCA protocol
bySample <- lapply(bySample, function (x) {
  x <- x %>% NormalizeData(verbose=F) %>%
    FindVariableFeatures(nfeatures = nfeatures,
                         verbose=F) %>%
    ScaleData(verbose=F) %>%
    RunPCA(verbose=F)
  return(x)
})

commonFeatures <- SelectIntegrationFeatures(bySample,
                                            verbose=F)

smanchors <- FindIntegrationAnchors(bySample,
                                    anchor.features = commonFeatures,
                                    verbose=F)

bySample.cca <- IntegrateData(anchorset = smanchors,
                              verbose=F)

DefaultAssay(bySample.cca) <- "integrated"

# Run the standard workflow for clustering and visualization of integrated data
bySample.cca <- bySample.cca %>%
  ScaleData(verbose = FALSE) %>%
  RunPCA(verbose = FALSE) %>%
  FindNeighbors(reduction = "pca",
                dims = seq(npcs),
                force.recalc = T,
                verbose=F) %>%
  FindClusters(resolution = ress,
               verbose=F) %>%
  RunUMAP(reduction = "pca",
          dims = seq(npcs),
          verbose=F) %>%
  RunTSNE(reduction = "pca",
          dims = seq(npcs),
          perplexity=100,
          verbose=F)

# Reannotate clusters
bySample.cca@meta.data <- bySample.cca@meta.data %>%
  mutate(across(starts_with("integrated_"),.fns = function (x){paste0("C",x)})) %>%
  mutate(across(starts_with("integrated_"),.fns = function (x){factor(x)})) %>%
  mutate(across(starts_with("integrated_"),.fns = function (x){factor(x,levels=mixedsort(levels(x)))}))

DefaultAssay(bySample.cca) <- "RNA"

saveRDS(bySample.cca,file = file.path(outputPath,paste(cellSet,subsetName,"seurat","rds",sep = ".")))

bySample.cca
```

```{r}
clustering <- "integrated_snn_res.0.1"
```

```{r}
p1 <- DimPlot(bySample.cca,
              reduction = "tsne",
              group.by = "SampleName",
              cols = color.list)
p2 <- DimPlot(bySample.cca,
              reduction = "tsne",
              group.by = clustering,
              cols = color.list,
              label = T)
p1 + p2
```

## Remove Doublets

```{r}
sce <- Seurat::as.SingleCellExperiment(bySample.cca, assay = "RNA")
clusters <- quickCluster(sce)
sce <- computeSumFactors(sce, clusters=clusters)
summary(sizeFactors(sce))
dbs <- computeDoubletDensity(
  sce
)
sce <- scDblFinder(sce)

bySample.cca@meta.data$doublets <- colData(sce)[colnames(bySample.cca),"scDblFinder.class"]
```

```{r}
DimPlot(bySample.cca,
        reduction = "tsne",
        group.by = "doublets",
        cols = c("grey70","red")
        )
```

```{r}
bySample.cca@meta.data %>%
  group_by(integrated_snn_res.0.1) %>%
  summarise(PctDoublets = sum(doublets == "doublet")/n())
```

```{r}
myCells <- colnames(bySample.cca)[!(bySample.cca$doublets == "doublet" | bySample.cca$integrated_snn_res.0.1 %in% c("C8","C11","C12"))]

sc <- sc[,myCells]
sc
```

## Reclustering without doublets

```{r}
cellSet <- "Stromal"
subsetName <- "IntegratedFilteredDoublets"
```

```{r}
nfeatures <- 2000
ress <- c(0.1)
npcs <- 20

bySample <- SplitObject(sc, split.by = "SampleName")

# Integration via CCA protocol
bySample <- lapply(bySample, function (x) {
  x <- x %>% NormalizeData(verbose=F) %>%
    FindVariableFeatures(nfeatures = nfeatures,
                         verbose=F) %>%
    ScaleData(verbose=F) %>%
    RunPCA(verbose=F)
  return(x)
})

commonFeatures <- SelectIntegrationFeatures(bySample,
                                            verbose=F)

smanchors <- FindIntegrationAnchors(bySample,
                                    anchor.features = commonFeatures,
                                    dims = seq(npcs),
                                    verbose=F)

bySample.cca <- IntegrateData(anchorset = smanchors,
                              dims = seq(npcs),
                              verbose=F)

DefaultAssay(bySample.cca) <- "integrated"

# Run the standard workflow for clustering and visualization of integrated data
bySample.cca <- bySample.cca %>%
  ScaleData(verbose = FALSE) %>%
  RunPCA(verbose = FALSE) %>%
  FindNeighbors(reduction = "pca",
                dims = seq(npcs),
                force.recalc = T,
                verbose=F) %>%
  FindClusters(resolution = ress,
               verbose=F) %>%
  RunUMAP(reduction = "pca",
          dims = seq(npcs),
          verbose=F) %>%
  RunTSNE(reduction = "pca",
          dims = seq(npcs),
          perplexity=100,
          verbose=F)

# Reannotate clusters
bySample.cca@meta.data <- bySample.cca@meta.data %>%
  mutate(across(starts_with("integrated_"),.fns = function (x){paste0("C",x)})) %>%
  mutate(across(starts_with("integrated_"),.fns = function (x){factor(x)})) %>%
  mutate(across(starts_with("integrated_"),.fns = function (x){factor(x,levels=mixedsort(levels(x)))}))

saveRDS(bySample.cca,file = file.path(outputPath,paste(cellSet,subsetName,"seurat","rds",sep = ".")))

bySample.cca
```
```{r}
clustering <- "integrated_snn_res.0.1"
```

```{r}
p1 <- DimPlot(bySample.cca,
              reduction = "tsne",
              group.by = "SampleName",
              cols = color.list)
p2 <- DimPlot(bySample.cca,
              reduction = "tsne",
              group.by = clustering,
              cols = color.list,
              label = T)
p1 + p2
```

## Get Stromal Cell Type Modules from Scaden Stromal Single Cell.

[Baryawno et al. Cell 2019](https://doi.org/10.1016/j.cell.2019.04.040)

```{r}
scaden <- readRDS(file.path(dataPath,"Scadden_niche_original.Rds"))
```

```{r}
p1 <- DimPlot(scaden, 
              reduction = "tsne", 
              group.by = "Scadden_annot", 
              cols = color.list)
p2 <- DimPlot(scaden, 
              reduction = "tsne", 
              group.by = "cell_type", 
              cols = color.list)
p1 + p2
```

### Scaden markers based on Cell Type

```{r}
Idents(scaden) <- "cell_type"
markersScaden <- scaden %>%
  FindAllMarkers(
    assay = "data",
    test.use = "wilcox",
    slot = "data",
    only.pos = T,
    verbose = F)

table(markersScaden[markersScaden$p_val_adj < 0.01,"cluster"])
```

#### HeatMap of Top 100 Scaden Cell Type Markers

```{r}
top100 <- markersScaden %>%
  filter(p_val_adj < 0.01) %>%
    group_by(cluster) %>%
    top_n(n = 100, wt = avg_log2FC)
```

```{r}
scaden <- ScaleData(scaden,
                    features = unique(sort(c(VariableFeatures(scaden),top100$gene))),
                    verbose = F)
DoHeatmap(scaden, features = top100$gene) + NoLegend()
```

#### Top 20 Scaden Markers DotPlot

```{r fig.width=12}
ntop <- 10
pval <- 0.01

topMarkers <- markersScaden %>%
  filter(p_val_adj < pval) %>%
  group_by(cluster) %>%
  top_n(ntop, avg_log2FC) %>%
  ungroup() %>%
  dplyr::select(gene) %>%
  distinct()

DotPlot(
  scaden,
  features = rev(topMarkers)
  ) +
  theme(axis.text.x = element_text(angle = 45,hjust = 1))
```

### Select Cell Type Module genes from markers

Gene modules selected from the top 200 marker genes at an adjusted pval < 0.01, a mÃ­nimum logFC of 0.25, present in our Stromal dataset and not shared between cell types.

```{r}
celltypeMarkers <- markersScaden %>%
  filter(p_val_adj < 0.01) %>%
  filter(avg_log2FC > 0.25) %>%
  filter(gene %in% rownames(bySample.cca)) %>%
  group_by(cluster) %>%
  top_n(n = 200, wt = avg_log2FC) %>%
  filter(gene %in% rownames(bySample.cca)) %>%
  dplyr::select(cluster,gene)

whichDup <- unique(sort(celltypeMarkers$gene[which(duplicated(celltypeMarkers$gene))]))
celltypeMarkers <- celltypeMarkers %>% filter(!gene %in% whichDup)

table(celltypeMarkers$cluster)

scaden <- ScaleData(scaden,
                    features = unique(sort(c(VariableFeatures(scaden),celltypeMarkers$gene))),
                    verbose = F)
DoHeatmap(scaden, features = celltypeMarkers$gene) + NoLegend()
```

```{r}
moduleGeneList <- celltypeMarkers %>%
  group_by(cluster) %>%
  group_split()
moduleGeneList <- as.list(moduleGeneList)
names(moduleGeneList) <- levels(factor(celltypeMarkers$cluster))
```

### Scaden Cell Type Module Scores

```{r}
bySample.cca@meta.data <- bySample.cca@meta.data %>% dplyr::select(-contains("Module"))
for (i in names(moduleGeneList)) {
  bySample.cca <- AddModuleScore(bySample.cca
                             ,features = list(i=moduleGeneList[[i]]$gene)
                             ,name =  paste0(i,".RNAModule")
                             ,assay = "RNA"
                             )
}
```

### Cell Type Module Scores Plots

```{r}
df <- Embeddings(bySample.cca,reduction = "tsne")
dims <- colnames(df)
df <- cbind(df,bySample.cca@meta.data %>% dplyr::select(contains(".RNAModule1")))
```

```{r}
df %>%
  pivot_longer(cols = contains("RNAModule1"), names_to = "Module", values_to = "Score") %>%
  mutate(Module=sub(".RNAModule1","",Module)) %>%
  ggplot(aes_string(x=dims[1],y=dims[2],color="Score")) +
  geom_point() +
  scale_color_gradientn(colors = colorRampPalette(c("grey","orange","red"))(3),name="Log(NormCounts)") +
  facet_wrap("Module") +
  theme_classic()
```

### Module Asingments to Clusters

Cluster assigned to the maximum enrichment module score +- 0.02. Some clusters may score two modules or more as maximum.

Analysis based on WT transcriptome only.

```{r}
clustering <- "integrated_snn_res.0.1"
bySample.cca@meta.data %>%
  filter(SampleName == "CD63_WT") %>%
  dplyr::select(contains(c(clustering,"RNAModule1"))) %>%
  pivot_longer(cols = contains("RNAModule1"),names_to = "Module",values_to = "Score") %>%
  dplyr::rename(Cluster=clustering) %>%
  group_by(Module) %>%
  summarise(Score=scale(Score,center = T,scale = T),Cluster=Cluster) %>%
  group_by(Cluster,Module) %>%
  summarise(AverageScore=mean(Score)) %>%
  group_by(Cluster) %>%
  summarise(Module=Module,AverageScoreScaled=scale(AverageScore),MaxEnrichment=(AverageScore >= (max(AverageScore)-0.02))) %>%
  mutate(Module=sub(".RNAModule1","",Module)) %>%
  mutate(plotBorder=ifelse(MaxEnrichment,1.5,0)) %>%
  ggplot() +
  geom_point(aes_string(x="Module",y="Cluster",fill="AverageScoreScaled",color="MaxEnrichment", stroke = "plotBorder"),size=6,shape=21) +
  scale_fill_gradient(low = "grey90",high = "blue") +
  scale_color_manual(values = c(NA,"red"))
```

```{r message=FALSE, warning=FALSE}
# Assign Sommerkamp cell type to clusters using WT info
cellTypeAnnot <- bySample.cca@meta.data %>%
  filter(SampleName == "CD63_WT") %>%
  dplyr::select(contains(c(clustering,"RNAModule1"))) %>%
  pivot_longer(cols = contains("RNAModule1"),names_to = "Module",values_to = "Score") %>%
  dplyr::rename(Cluster=clustering) %>%
  group_by(Module) %>%
  summarise(Score=scale(Score,center = T,scale = T),Cluster=Cluster) %>%
  group_by(Cluster,Module) %>%
  summarise(AverageScore=mean(Score)) %>%
  group_by(Cluster) %>%
  summarise(Module=Module,AverageScoreScaled=scale(AverageScore),MaxEnrichment=(AverageScore >= (max(AverageScore)-0.02))) %>%
  mutate(Module=sub(".RNAModule1","",Module)) %>%
  mutate(Selected=ifelse(MaxEnrichment,1,0)) %>%
  filter(Selected == 1) %>%
  group_by(Cluster) %>%
  summarise(CellType1 = paste(Module,collapse="_"))
```

```{r}
bySample.cca@meta.data <- bySample.cca@meta.data %>%
  left_join(cellTypeAnnot, by = c("integrated_snn_res.0.1" = "Cluster")) %>% as.data.frame()
rownames(bySample.cca@meta.data) = colnames(bySample.cca)
```

### Cell Type By Cluster

```{r}
p1 <- DimPlot(bySample.cca,
              reduction = "tsne",
              group.by = clustering,
              cols = color.list,
              label = T) + theme(legend.position = "none")
p2 <- DimPlot(bySample.cca,
              reduction = "tsne",
              group.by = "CellType1", 
              cols = color.list,
              label = F)
p1 + p2
```

### Cell Type By Condition

```{r}
DimPlot(bySample.cca,
        reduction = "tsne",
        group.by = "CellType1",
        cols = color.list,
        split.by = "SampleName")
```


### Cell Type Proportions

```{r}
bySample.cca@meta.data %>%
  ggplot(aes(x="",fill=CellType1)) +
  geom_bar(stat="count",position="fill") +
  coord_polar("y",start = 0) +
  scale_fill_manual(values = color.list) +
  facet_wrap("SampleName") +
  theme(axis.line = element_blank(),
        axis.text = element_blank(),
        axis.title = element_blank(),
        panel.background = element_blank())
```

```{r}
bySample.cca@meta.data %>%
  group_by(SampleName) %>%
  dplyr::count(
    CellType1
  ) %>%
  pivot_wider(id_cols = SampleName, names_from = CellType1, values_from = n)
```

## Marker Genes for each cell type

Based on WT transcriptomes.

```{r}
clustering <- "CellType1"

Idents(bySample.cca) <- clustering

minPct <- 30
pval <- 0.01
useAssay <- "RNA"
method <- "wilcox"

DefaultAssay(bySample.cca) <- useAssay
bySample.cca.WT <- subset(bySample.cca, SampleName == "CD63_WT")
```

```{r }
markers <- bySample.cca.WT %>%
  FindAllMarkers(
    assay = useAssay
    , slot = "data"
    , min.pct = minPct/100
    , return.thresh = pval
    , test.use = method
    , verbose = F
    , only.pos = T
  )
```

```{r }
markers %>%
  filter(p_val_adj < pval) %>%
  group_by(cluster) %>%
  dplyr::count()
```

### Top 30 Markers Heatmap

```{r fig.asp=1.4}
ntop <- 30

topMarkers <- markers %>% 
  filter(p_val_adj < pval) %>%
  group_by(cluster) %>%
  top_n(ntop, avg_log2FC)

bySample.cca <- bySample.cca %>%
  ScaleData(features = c(VariableFeatures(.),topMarkers$gene),
            assay = useAssay,
            verbose = F)

bySample.cca %>%
  DoHeatmap(assay = useAssay,
            features = topMarkers$gene,
            group.colors = color.list)
```

### Top 9 Markers Expression

```{r fig.width=12,fig.asp=0.7}
ntop <- 9
for (myCluster in levels(markers$cluster)) {
  topMarkers <- markers %>% 
  filter(p_val_adj < pval & cluster == myCluster) %>%
  top_n(ntop, avg_log2FC)
  
  print(bySample.cca %>%
    FeaturePlot(features = topMarkers$gene,
                reduction = "tsne") +
      NoLegend() +
      plot_annotation(
        title = paste("Markers for cell type:",myCluster
                      )
))
}
```

### Top 10 Markers DotPlot

```{r fig.width=12}
ntop <- 10
pvalcut <- 0.01

topMarkers <- markers %>%
  filter(p_val_adj < pval) %>%
  group_by(cluster) %>%
  top_n(ntop, avg_log2FC) %>%
  ungroup() %>%
  dplyr::select(gene) %>%
  distinct()

DotPlot(
  bySample.cca,
  assay = useAssay,
  group.by = clustering,
  features = rev(topMarkers)
  ) +
  theme(axis.text.x = element_text(angle = 45,hjust = 1))
```

## Differences between Conditions for each Cell Type

Within each Cell Type we have tested the differences between KO and WT conditions.

```{r}
tde.files <- NULL
contrastByCond <- list()
useAssay <- "RNA"
clustering <- "CellType1"
testUse <- "wilcox"
minPct <- 0.1

cond1 <- "CD63_KO"
cond2 <- "CD63_WT"

DefaultAssay(bySample.cca) <- useAssay

bySample.cca <- bySample.cca %>%
  SetIdent(value = paste(bySample.cca$CellType1,bySample.cca$SampleName,sep="."))

deGenes <- list()
for (myCluster in levels(factor(bySample.cca$CellType1))) {
        
        ident1 <- paste(myCluster,cond1,sep=".")
        ident2 <- paste(myCluster,cond2,sep=".")
        
        deGenes[[myCluster]] <- bySample.cca %>%
          FindMarkers(
            assay = useAssay
            , slot = "data"
            , ident.1 = ident1
            , ident.2 = ident2
            , min.pct = minPct
            , test.use = testUse
            , verbose = F
          ) %>%
          mutate(gene = rownames(.),
                 cluster = myCluster)
}

Idents(bySample.cca) <- clustering
```

```{r}
pval <- 0.01

deGenes <- do.call(rbind,deGenes)

deGenes %>%
  filter(p_val_adj < pval) %>%
  group_by(cluster) %>%
  dplyr::count()
```

### Top 10 Differences DotPlot

```{r fig.width=12}
ntop <- 10
pvalcut <- 0.01

topDeGenes <- deGenes %>%
  filter(p_val_adj < pval) %>%
  group_by(cluster) %>%
  top_n(ntop, avg_log2FC) %>%
  ungroup() %>%
  dplyr::select(gene) %>%
  distinct()

DotPlot(
  bySample.cca,
  assay = "RNA",
  group.by = clustering,
  features = rev(topDeGenes),
  split.by = "SampleName"
  ) +
  theme(axis.text.x = element_text(angle = 45,hjust = 1))
```

## Il1B and Il1rn Expression

```{r}
df <- Embeddings(bySample.cca,reduction = "tsne")
dnames <- colnames(df)
df <- cbind(df,bySample.cca@meta.data)
df <- cbind(df,FetchData(bySample.cca,vars = c("Il1b","Il1rn")))
```

```{r}
df %>%
  arrange(Il1b) %>%
  ggplot(aes_string(x=dnames[1],y=dnames[2])) +
  geom_point(aes_string(color="Il1b")) +
  facet_wrap("SampleName") +
  ggtitle("Il1b") +
  scale_color_gradientn(colors = colorRampPalette(c("grey","orange","red"))(3),name="Log(NormCounts)") +
  theme_classic()
```

```{r}
df %>%
  arrange(Il1rn) %>%
  ggplot(aes_string(x=dnames[1],y=dnames[2])) +
  geom_point(aes_string(color="Il1rn")) +
  facet_wrap("SampleName") +
  ggtitle("Il1rn") +
  scale_color_gradientn(colors = colorRampPalette(c("grey","orange","red"))(3),name="Log(NormCounts)") +
  theme_classic()
```

```{r fig.width=14, fig.asp=0.5}
df %>%
  mutate(Il1Exp = case_when(
    Il1b>0 & Il1rn == 0 ~ "Il1b only",
    Il1b==0 & Il1rn > 0 ~ "Il1rn only",
    Il1b>0 & Il1rn > 0 ~ "Il1b and Il1rn")) %>%
 ggplot(aes(x="",fill=Il1Exp)) +
  geom_bar(stat="count",position="fill") +
  coord_polar("y",start = 0) +
  scale_fill_manual(values = color.list) +
  facet_wrap(c("SampleName","CellType1"),ncol = 6) +
  theme(axis.line = element_blank(),
        axis.text = element_blank(),
        axis.title = element_blank(),
        panel.background = element_blank())
```

## Il1b vs Il1rn

```{r}
p <- df %>%
  filter(SampleName == "CD63_WT")  %>%
  mutate(Il1Exp = case_when(
    Il1b>0 & Il1rn == 0 ~ "Il1b only",
    Il1b==0 & Il1rn > 0 ~ "Il1rn only",
    Il1b>0 & Il1rn > 0 ~ "Il1b and Il1rn")) %>%
  ggplot(aes(x=Il1rn,y=Il1b)) +
  ggtitle("Stromal WT Cells") +
  geom_point() +
  theme_classic()

ggMarginal(p,type = "densigram")
```


## Violin plots on Selected Genes

```{r}
selectedGenes <- list()
selectedGenes[["Fb_MSC"]] <- c("Nr1d1","Zbtb16","Nfia","Hspg2","Bmp4","Col3a1","Fbn1","Twist1","Spp1")
selectedGenes[["Sinu"]] <- sort(unique(c("Il6","Il1b","Csf1","Il6","Il1b","Osm","Ccl9","Ccl6","Nlrp3","Csf1","Ccl4","Il6","Il1b","Osm","Ccl9","Ccl6","Nlrp3")))
```

```{r}
df <- FetchData(bySample.cca,vars = unlist(selectedGenes))
df <- cbind(df,bySample.cca@meta.data)
df <- df %>% mutate(SampleName=factor(SampleName,levels = c("CD63_WT","CD63_KO")))
```

### Fibroblasts

```{r}
df %>%
  filter(CellType1 %in% c("Fibroblasts")) %>%
  pivot_longer(cols = selectedGenes$Fb_MSC,names_to = "genes",values_to = "expression") %>%
  ggplot(aes_string(y="expression",x="SampleName",color="SampleName",fill="SampleName")) +
  geom_violin(stat = "ydensity",alpha=0.2) +
  # geom_jitter(alpha=1) +
  scale_color_manual(values = c("grey40","#a80a0b")) +
  scale_fill_manual(values = c("grey40","#a80a0b")) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("Expression Level log(NormCounts+1)") +
  ggtitle("Fibroblast") +
  facet_wrap("genes",nrow = 1) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90),legend.position = "none")
```

### MSCs

```{r}
df %>%
  filter(CellType1 %in% c("MSCs")) %>%
  pivot_longer(cols = selectedGenes$Fb_MSC,names_to = "genes",values_to = "expression") %>%
  ggplot(aes_string(y="expression",x="SampleName",color="SampleName",fill="SampleName")) +
  geom_violin(stat = "ydensity",alpha=0.2) +
  # geom_jitter(alpha=1) +
  scale_color_manual(values = c("grey40","#a80a0b")) +
  scale_fill_manual(values = c("grey40","#a80a0b")) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("Expression Level log(NormCounts+1)") +
  ggtitle("MSCs") +
  facet_wrap("genes",nrow = 1) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90),legend.position = "none")
```

### Sinusoidal

```{r}
df %>%
  filter(CellType1 %in% c("Sinusoidal")) %>%
  pivot_longer(cols = selectedGenes$Sinu,names_to = "genes",values_to = "expression") %>%
  ggplot(aes_string(y="expression",x="SampleName",color="SampleName",fill="SampleName")) +
  geom_violin(stat = "ydensity",alpha=0.2) +
  # geom_jitter(alpha=1) +
  scale_color_manual(values = c("grey40","#a80a0b")) +
  scale_fill_manual(values = c("grey40","#a80a0b")) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("Expression Level log(NormCounts+1)") +
  ggtitle("Sinusoidal") +
  facet_wrap("genes",nrow = 1) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90),legend.position = "none")
```

# Session Info

```{r}
sessionInfo()
```


